### DISCUSSION TITLE
Method references

### DISCUSSION THREAD

**Discussion by Vendetta8247** [iBotPeaches/Apktool#2630] [Q&A]
Hi! I've stumbled upon an issue where dex is being overflowed.
Manual checking all .smali files led me to a result of actually 50k methods, not 65k.
Then I checked the Android studio APK Analyzer and found that there is a thing called referenced methods. Didn't meet this terminology before so wasn't sure what to do with it.
I have checked one of the files that is supposed to feature this referenced method and found that it is absent there.

So my question is: am I missing something about the reference methods? They are present in dex files afaiu but aren't present in the smali files. If I'm in the wrong spot - I'm sorry but not sure there is a better place to ask this :)

**Comment by Comnir** [comment]
Yes, the limit is on the number of methods referenced as explained on [this documentation page](https://developer.android.com/studio/build/multidex).
I guess you are changing some smali files or add files and then encounter this error during build. 

I'm not sure about how it is counted exactly, but it is possible to avoid this failure by copying the changed/new smali files to a new folder before building.

**Reply by Vendetta8247** [reply]
Yes, this is exactly what I decided to go for but I wasn't sure if it was impacting build times or not (didn't bother to check yet as it was less important for now). Yet still this information is present in dex files so I'm particularly curious if it can be extracted somehow (apktool.yml?)
